src/modules/netinstall/NetInstallViewStep.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ *   Copyright 2016, Luca Giambonini <almack@chakraos.org>
+ *                   Lisa Vitolo <shainer@chakraos.org>
+ *
+ *   Calamares is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   Calamares is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with Calamares. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "NetInstallViewStep.h"
+
+#include "JobQueue.h"
+#include "GlobalStorage.h"
+#include "utils/Logger.h"
+
+#include "NetInstallPage.h"
+
+CALAMARES_PLUGIN_FACTORY_DEFINITION( NetInstallViewStepFactory, registerPlugin<NetInstallViewStep>(); )
+
+NetInstallViewStep::NetInstallViewStep( QObject* parent )
+    : Calamares::ViewStep( parent )
+    , m_widget( new NetInstallPage() )
+    , m_nextEnabled( true )
+{
+    emit nextStatusChanged( true );
+    connect( m_widget, &NetInstallPage::checkReady,
+             this, &NetInstallViewStep::nextStatusChanged );
+}
+
+
+NetInstallViewStep::~NetInstallViewStep()
+{
+    if ( m_widget && m_widget->parent() == nullptr )
+        m_widget->deleteLater();
+}
+
+
+QString
+NetInstallViewStep::prettyName() const
+{
+    return tr( "NetInstall" );
```

Comment:
This is the user-visible name, maybe something more descriptive? "Select packages" or "Package selection" or somesuch?